lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -724,6 +769,47 @@ void ProxyRenderDelegate::_UpdateSceneDelegate()
     }
 }
 
+#ifdef MAYA_HAS_DISPLAY_LAYER_API
+void ProxyRenderDelegate::_UpdateDisplayLayers()
+{
+    // make sure we're listening to every display layer for attribute changes
+    MObject displayLayerManagerHandle = MFnDisplayLayerManager::currentDisplayLayerManager();
+    MFnDisplayLayerManager displayLayerManagerFn(displayLayerManagerHandle);
+    MObjectArray newDisplayLayers(displayLayerManagerFn.getAllDisplayLayers());
+    bool sameLayers = newDisplayLayers.length() == _mayaDisplayLayers.length();
+    for(unsigned int i=0; i<_mayaDisplayLayers.length() && sameLayers; ++i) {
+        sameLayers = _mayaDisplayLayers[i] == newDisplayLayers[i];
+    }
+
+    if (!sameLayers) {
+        // remove all the callbacks from _mayaDisplayLayers
+        TF_VERIFY(_mayaDisplayLayers.length() == _mayaDisplayLayerCallbacks.size());
+        for(const auto& callbackId : _mayaDisplayLayerCallbacks)
+            MMessage::removeCallback(callbackId);
+
+        _mayaDisplayLayers = newDisplayLayers;
+        _mayaDisplayLayerCallbacks.clear();
+
+        // add all the callbacks to _mayaDisplayLayers;
+        for(unsigned int i=0; i<_mayaDisplayLayers.length(); ++i) {
+            _mayaDisplayLayerCallbacks.push_back(MNodeMessage::addNodeDirtyCallback(_mayaDisplayLayers[i], DisplayLayerDirtyCB, this, nullptr));
+        }
+    }
+
+    HdChangeTracker& changeTracker = _renderIndex->GetChangeTracker();
+    if (_displayLayerMembershipChanged) {
+        auto&            rprims = _renderIndex->GetRprimIds();
+        for (auto path : rprims) {
+            fprintf(stderr, "dirtied %s\n", path.GetText());
```

Comment:
Did you mean to leave this in?

Reply:
No, I had some local changes I forgot to commit. Sorry!

Reply:
I still see the fprintf